prepare_run.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+import subprocess
+import argparse
+
+
+# Usage: ./prepare_run.py NAO-ONT-20240519-practice
+# This script batches pod5 files from a delivery directory and prepares them for processing
+
+def list_s3_files(S3_BUCKET, delivery):
+    """List files in S3 with the given prefix"""
+    pod5_dir = f"{delivery}/pod5/"
+    cmd = ["aws", "s3", "ls", f"s3://{S3_BUCKET}/{pod5_dir}"]
+
+    result = subprocess.run(cmd, stdout=subprocess.PIPE, text=True, check=True)
+
+    files = []
+    for line in result.stdout.strip().split('\n'):
+        if not line:
+            continue
+        parts = line.split()
+        if len(parts) >= 4:  # Standard S3 ls output format
+            file_name = ' '.join(parts[3:])
+            size = int(parts[2])
+            files.append((file_name, size))
+
+    return files
+
+
+def copy_file_to_s3(S3_BUCKET, pod5_file, delivery, batch_id):
+    """Copy a file from one S3 location to another"""
+
+    origin = f"s3://{S3_BUCKET}/{delivery}/pod5/{pod5_file}"
+    destination = f"s3://{S3_BUCKET}/{delivery}/batched/{batch_id}/{pod5_file}"
+
+    # Check if file already exists at destination
+    check_cmd = ["aws", "s3", "ls", destination]
+    result = subprocess.run(check_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    if result.returncode == 0:
+        print(f"File {pod5_file} already exists in batch {batch_id}, skipping...")
+        return
+
+    cmd = ["aws", "s3", "cp", origin, destination]
+
+    subprocess.run(cmd, check=True)
+
+def batch_input_files(files, BATCH_SIZE, max_files_per_batch=10):
+    """Group input files into batches based on size and count constraints"""
+    current_batch = []
+    current_batch_size = 0
+
+
+    for file_name, size in files:
+        # If adding this file would exceed batch constraints, yield current batch
+        if (current_batch_size + size > BATCH_SIZE or
+            len(current_batch) >= max_files_per_batch) and current_batch:
+
+            yield current_batch
+            current_batch = []
+            current_batch_size = 0
+
+        current_batch.append(file_name)
+        current_batch_size += size
+
+    # Yield the last batch if not empty
+    if current_batch:
+        yield current_batch
+
+def main():
+    BATCH_SIZE = (1024**3)  # 1GB
```

Comment:
better to call this `target_batch_size`, since we can produce batches smaller or larger than this but it's what we're going for